spec/system/admin/orders_spec.rb

```diff
@@ -269,17 +269,20 @@
       end
 
       it "displays non-empty cart orders" do
-        pending "issue #11120"
-
         # empty cart order does not appear in the results
         expect(page).not_to have_content order_empty.number
 
         # non-empty cart order, with bill- and ship-address appear in the results
         expect(page).to have_content order_not_empty.number
 
         # non-empty cart order, with no with bill- and ship-address appear in the results
+        expect(page).to have_content order_not_empty_no_address.number
 
-        # pending issue #11120
+        # And the same orders are displayed when sorting by name:
+        find("th a", text: "NAME").click
+
+        expect(page).to have_no_content order_empty.number
+        expect(page).to have_content order_not_empty.number
```

Comment:
This extends the spec. I kept it in the same block to save on execution time. There was another inner join in this case which was hiding orders without bill address.